src/server/src/main/java/io/cassandrareaper/storage/CassandraStorage.java

```diff
@@ -1565,6 +1579,60 @@ private static boolean withinRange(RepairSegment segment, Optional<RingRange> ra
     return !range.isPresent() || segmentIsWithinRange(segment, range.get());
   }
 
+  @Override
+  public Collection<DiagEventSubscription> getEventSubscriptions() {
+    return session.execute(getDiagnosticEventsPrepStmt.bind()).all().stream()
+        .map((row) -> createDiagEventSubscription(row))
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public Collection<DiagEventSubscription> getEventSubscriptions(String clusterName) {
+    return session.execute(getDiagnosticEventsPrepStmt.bind()).all().stream()
+        .map((row) -> createDiagEventSubscription(row))
+        .filter((subscription) -> clusterName.equals(subscription.getCluster()))
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public DiagEventSubscription getEventSubscription(UUID id) {
+    return createDiagEventSubscription(session.execute(getDiagnosticEventPrepStmt.bind(id)).one());
+  }
+
+  private static DiagEventSubscription createDiagEventSubscription(Row row) {
+    return new DiagEventSubscription(
+        Optional.of(row.getUUID("id")),
+        row.getString("cluster"),
+        Optional.of(row.getString("description")),
+        row.getList("include_nodes", String.class),
+        row.getList("events", String.class),
+        row.getBool("export_sse"),
+        row.getString("export_file_logger"),
+        row.getString("export_http_endpoint"));
+  }
+
+  @Override
+  public DiagEventSubscription addEventSubscription(DiagEventSubscription subscription) {
+    session.execute(saveDiagnosticEventPrepStmt.bind(
+        subscription.getId(),
+        subscription.getCluster(),
+        subscription.getDescription(),
+        subscription.getIncludeNodes(),
+        subscription.getEvents(),
+        subscription.getExportSse(),
+        subscription.getExportFileLogger(),
+        subscription.getExportHttpEndpoint()));
+
+    return subscription;
+  }
+
+  @Override
+  public boolean deleteEventSubscription(UUID id) {
+    session.executeAsync(deleteDiagnosticEventPrepStmt.bind(id));
+    return false;
```

Comment:
This makes every REST call fail with a 500 error when trying to delete existing subscriptions.

Reply:
Since we need to refresh the UI to get an updated view of the subscriptions, I think we need to use a synchronous query instead, at `LOCAL_QUORUM`.

Reply:
Going through the UI does not need strict consistency. (You could just be on a page that lists a subscription that has been deleted elsewhere.)
It doesn't really need to be sync either, durability of the delete is not critical. 

I presume the REST call was failing only because of the `return false`, thanks for fixing that.